### PR TITLE
Add `predictVoucher(..)` & `isVoucher(..)` functions

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: test/VoucherHub.test.ts [hunk 1]
```diff
@@ -996,10 +996,29 @@ describe('VoucherHub', function () {
         });
     });
 
+    describe('Is voucher', function () {
```
Comment (zguesmi) [inline test/VoucherHub.test.ts:999]: Let's add a test where a voucher contract is deployed manually not through the VoucherHub and check that `isVoucher`, and `getVoucher` return respectively `false` and `0x0`.

**Review by gfournieriExec — APPROVED** [review]
Thanks

File: test/VoucherHub.test.ts (hunk 1 above)
Comment (jeremyjams) [reply test/VoucherHub.test.ts:999]: As discussed, a voucher contract deployed manually won't be able to land on a legitimate voucher address

**Review by gfournieriExec — APPROVED** [review]

File: contracts/VoucherHub.sol [hunk 2]
```diff
@@ -327,19 +327,37 @@ contract VoucherHub is
...
+    function getVoucher(address owner) external view returns (address voucherAddress) {
+        voucherAddress = predictVoucher(owner);
         return voucherAddress.code.length > 0 ? voucherAddress : address(0);
     }
 
+    /**
+     * Predict the address of the (created or not) voucher for a given owner.
+     * @param owner The owner of the voucher.
+     */
+    function predictVoucher(address owner) public view returns (address) {
```
Comment (zguesmi) [inline contracts/VoucherHub.sol:352]:
```suggestion
    function predictVoucherAddress(address owner) public view returns (address) {
```

Comment (jeremyjams) [reply contracts/VoucherHub.sol:352]: I dont think we want to update `getVoucher()`..
https://github.com/p00ls/contracts/blob/master/contracts/tokens/P00lsRegistryBase.sol#L60-L66

**Review by zguesmi — APPROVED** [review]

File: contracts/VoucherHub.sol (hunk 2 above)
Comment (jeremyjams) [reply contracts/VoucherHub.sol:352]: Alright :+1:

**Pull request merged**